src/adagrams.js

```diff
@@ -1,15 +1,132 @@
+// helper function to call for range for letterBank
+function range(stop) {
+  let result = [];
+  // creates a list of integers starting and 0 and ending at stop variable
+  for (let i = 0; i < stop; i += 1) {
+    result.push(i)
+  }
+  return result;
+};
+
+
 export const drawLetters = () => {
-  // Implement this method for wave 1
+  const letterPool = {
+    'A': 9, 'B': 2, 'C': 2, 'D': 4, 'E': 12, 'F': 2,
+    'G': 3, 'H': 2, 'I': 9, 'J': 1, 'K': 1, 'L': 4,
+    'M': 2, 'N': 6, 'O': 8, 'P': 2, 'Q': 1, 'R': 6,
+    'S': 4, 'T': 6, 'U': 4, 'V': 2, 'W': 2, 'X': 1,
+    'Y': 2, 'Z': 1
+  };
+
+  let returnList = [];
+  let letterBank = [];
+
+  // creates letter bank with num of letters equal to letterPool values
+  for (let [key,value] of Object.entries(letterPool)) {
+    for (value in range(value)) {
+      letterBank.push(key);
+    }
+  }
+
+  // draws 10 random letters from letterBank
+  while (returnList.length < 10) {
+    let letter = letterBank[Math.floor(Math.random()* letterBank.length)];
+    returnList.push(letter);
+    // grabs index of letter from letterBank and removes that letter at the index
+    let index = letterBank.indexOf(letter);
+    letterBank.splice(index,1);
+  }
```

Comment:
Rather than creating two separate scopes, you could have created a for loop that ran 10 times and combined this `while loop` and `for/of` into one. With a for loop you could have used i as an index and avoided the additional loop. Remember we should keep the algorithmic complexity in mind.

Also, Prefer `const` for the loop variable in a `for/of` loop.

Remember to keep in mind where certain built-in function calls might be hiding a loop, and try to avoid using those within an outer loop (things like splice or includes). Even though the small size of the data in this means the absolute performance is still fast, we should keep the algorithmic complexity in mind. Imagine what would happen if we relaxed the data restrictions to use larger tile sets or hands.